tests/test_flexible_unet.py

```diff
@@ -0,0 +1,241 @@
+)
+CASES_3D = make_shape_cases(
+    models=[SEL_MODELS[0]],
+    spatial_dims=[3],
+    batches=[1],
+    pretrained=[False],
+    in_channels=3,
+    num_classes=10,
+    norm="batch",
+)
+
+# varying num_classes and in_channels
+CASES_VARIATIONS = []
+
+# change num_classes test
+# 20 classes
+# 2D
+CASES_VARIATIONS.extend(
+    make_shape_cases(
+        models=SEL_MODELS, spatial_dims=[2], batches=[1], pretrained=[False, True], in_channels=3, num_classes=20
+    )
+)
+# 3D
+CASES_VARIATIONS.extend(
+    make_shape_cases(
+        models=[SEL_MODELS[0]], spatial_dims=[3], batches=[1], pretrained=[False], in_channels=3, num_classes=20
+    )
+)
+
+# change in_channels test
+# 1 channel
+# 2D
+CASES_VARIATIONS.extend(
+    make_shape_cases(
+        models=SEL_MODELS, spatial_dims=[2], batches=[1], pretrained=[False, True], in_channels=1, num_classes=10
+    )
+)
+# 8 channel
+# 2D
+CASES_VARIATIONS.extend(
+    make_shape_cases(
+        models=SEL_MODELS, spatial_dims=[2], batches=[1], pretrained=[False, True], in_channels=8, num_classes=10
+    )
+)
+# 3D
+CASES_VARIATIONS.extend(
+    make_shape_cases(
+        models=[SEL_MODELS[0]], spatial_dims=[3], batches=[1], pretrained=[False], in_channels=1, num_classes=10
+    )
+)
+
+# change input shape test
+# 96
+# 2D 96x96 input
+CASES_VARIATIONS.extend(
+    make_shape_cases(
+        models=SEL_MODELS,
+        spatial_dims=[2],
+        batches=[1],
+        pretrained=[False, True],
+        in_channels=3,
+        num_classes=10,
+        input_shape=96,
+    )
+)
+# 2D 64x64 input
+CASES_VARIATIONS.extend(
+    make_shape_cases(
+        models=SEL_MODELS,
+        spatial_dims=[2],
+        batches=[1],
+        pretrained=[False, True],
+        in_channels=3,
+        num_classes=10,
+        input_shape=64,
+    )
+)
+
+# 3D 32x32x32 input
+CASES_VARIATIONS.extend(
+    make_shape_cases(
+        models=SEL_MODELS,
+        spatial_dims=[2],
+        batches=[1],
+        pretrained=[False],
+        in_channels=3,
+        num_classes=10,
+        input_shape=32,
+    )
+)
+
+# 3D 64x64x64 input
+CASES_VARIATIONS.extend(
+    make_shape_cases(
+        models=SEL_MODELS,
+        spatial_dims=[2],
+        batches=[1],
+        pretrained=[False],
+        in_channels=3,
+        num_classes=10,
+        input_shape=64,
+    )
+)
+
+# pretrain weight verified
+CASES_PRETRAIN = [
+    (
+        {
+            "in_channels": 3,
+            "out_channels": 10,
+            "backbone": SEL_MODELS[0],
+            "pretrained": True,
+            "spatial_dims": 2,
+            "norm": ("batch", {"eps": 1e-3, "momentum": 0.01}),
+        },
+        {
+            "in_channels": 3,
+            "num_classes": 10,
+            "model_name": SEL_MODELS[0],
+            "pretrained": True,
+            "spatial_dims": 2,
+            "norm": ("batch", {"eps": 1e-3, "momentum": 0.01}),
+        },
+        ["_conv_stem.weight"],
+    )
+]
+
+
+class TestFLEXIBLEUNET(unittest.TestCase):
```

Comment:
could you help add skip_if_quick for the entire class https://github.com/Project-MONAI/MONAI/blob/6da25ae54996a7be8e06b9e23a18a25ea44d70f2/tests/test_efficientnet.py#L283

Reply:
Hi @wyli ,

Should we add the decorator to the entire class or only the `test_pretrain` function?

Thanks.

Reply:
The class, as the downloading might be slow and causes timeout for the premerge ci

Reply:
Thanks @wyli , I have added the @skip_if_quick to the whole class.